sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -351,6 +352,17 @@ public <T> T findById(Object id, Class<T> domainType, PartitionKey partitionKey)
     public <T> T findById(String containerName, Object id, Class<T> domainType) {
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(domainType, "domainType should not be null");
+        CosmosEntityInformation<?, ?> cosmosEntityInformation = CosmosEntityInformation.getInstance(domainType);
+        String containerPartitionKey = cosmosEntityInformation.getPartitionKeyFieldName();
+        if ("id".equals(containerPartitionKey) && id != null) {
+            try {
+                return findById(id, domainType, new PartitionKey(id));
+            } catch (CosmosAccessException e) {
+                LOGGER.warn(e.getMessage());
```

Comment:
This is a breaking change, as currently customers get `CosmosAccessException` and they must have a handling for it, in case if something goes bad. Now we are logging and returning null and not throwing the exception. We should not change this behavior.

Reply:
Ok, will reset on this.